test/framework/control_plane.go

```diff
@@ -244,6 +244,39 @@ func WaitForKubeadmControlPlaneMachinesToExist(ctx context.Context, input WaitFo
 	}, intervals...).Should(Equal(int(*input.ControlPlane.Spec.Replicas)))
 }
 
+// WaitForKubeadmControlPlaneMachinesToExistInput is the input for WaitForKubeadmControlPlaneMachinesToExist.
+type WaitForOneKubeadmControlPlaneMachineToExistInput struct {
+	Lister       Lister
+	Cluster      *clusterv1.Cluster
+	ControlPlane *controlplanev1.KubeadmControlPlane
+}
+
+// WaitForKubeadmControlPlaneMachineToExist will wait until all control plane machines have node refs.
+func WaitForOneKubeadmControlPlaneMachineToExist(ctx context.Context, input WaitForOneKubeadmControlPlaneMachineToExistInput, intervals ...interface{}) {
+	By("waiting for one control plane node to exist")
+	inClustersNamespaceListOption := client.InNamespace(input.Cluster.Namespace)
+	// ControlPlane labels
+	matchClusterListOption := client.MatchingLabels{
+		clusterv1.MachineControlPlaneLabelName: "",
+		clusterv1.ClusterLabelName:             input.Cluster.Name,
+	}
+
+	Eventually(func() (bool, error) {
+		machineList := &clusterv1.MachineList{}
+		if err := input.Lister.List(ctx, machineList, inClustersNamespaceListOption, matchClusterListOption); err != nil {
+			fmt.Println(err)
```

Comment:
Leftover?

Reply:
🙈

Reply:
Actually no, it's not leftover, this is the pattern used throughout. These fmt.Printlns should be replaced with a writer to ginkgo's log. I'm going to keep this and file an issue for improvement.